wasm2c/wasm-rt-impl.c

```diff
@@ -211,28 +257,30 @@ static void os_cleanup_signal_handler(void) {
     perror("sigaction failed");
     abort();
   }
-
-  if (sigaltstack(NULL, NULL) != 0) {
-    perror("sigaltstack failed");
-    abort();
-  }
-
-  free(g_alt_stack);
 }
 #endif
 
 #endif
 
 void wasm_rt_init(void) {
 #if WASM_RT_INSTALL_SIGNAL_HANDLER
-  if (!g_signal_handler_installed) {
-    g_signal_handler_installed = true;
-    os_install_signal_handler();
-  }
+#if !WASM_RT_USE_STACK_DEPTH_COUNT
+  assert(!g_alt_stack);
+  g_alt_stack = os_allocate_and_install_altstack();
```

Comment:
Nit: It may make sense to call `wasm_rt_thread_init` within here, as logically what we are saying is that the main thread gets has any thread-specific instantiation performed by default. Same with `wasm_rt_free`

Reply:
Yeah... I think in a world where `wasm_rt_init` returns the pointer (and `wasm_rt_free` takes it), that makes total sense. But if `g_alt_stack` only exists if certain preprocessor macros are set a certain way, then `wasm_rt_init` would have to condition the call to `wasm_rt_thread_init` similarly, and that feels weird -- why aren't we doing a thread init if `WASM_RT_USE_STACK_DEPTH_COUNT` is set?

So I think as long as `g_alt_stack` exists, `wasm_rt_init` has to take responsibility for knowing the details itself; it can't take advantage of the abstraction barrier.

Reply:
On further review... I got rid of g_alt_stack and just let the OS keep track of it.

Reply:
On further further view... I think I found a way to do this to satisfy everybody. wasm_rt_init() calls wasm_rt_init_thread() as you propose. `g_alt_stack` is still there but it's a thread-local mostly just used for safety (it gets cross-checked against the actual altstack).